test/std-test.cc

```diff
@@ -46,6 +48,25 @@ TEST(ranges_std_test, format_vector_path) {
 #endif
 }
 
+TEST(ranges_std_test, format_quote_path) {
+  // Test quote problem described in #3319.
+#ifdef __cpp_lib_filesystem
+  auto vec =
+      std::vector<std::filesystem::path>{"path1/file1.txt", "path2/file2.txt"};
+  EXPECT_EQ(fmt::format("{}", vec),
+            "[\"path1/file1.txt\", \"path2/file2.txt\"]");
+#  ifdef __cpp_lib_optional
+  auto o = std::optional<std::filesystem::path>("path/file.txt");
+  EXPECT_EQ(fmt::format("{}", o), "optional(\"path/file.txt\")");
+  EXPECT_EQ(fmt::format("{:?}", o), "optional(\"path/file.txt\")");
+#  endif
+#  ifdef __cpp_lib_variant
+  auto v = std::variant<std::filesystem::path, int>("path/file.txt");
+  EXPECT_EQ(fmt::format("{}", v), "variant(\"path/file.txt\")");
+#  endif
```

Comment:
I don't think we need these two because optional and variant use the same escaping mechanism as ranges.

Reply:
Done.